modules/effects/src/main/scala/dev/profunktor/redis4cats/commands.scala

```diff
@@ -41,4 +50,327 @@ object RedisCommands {
     def liftK[G[_]: Async: Log]: RedisCommands[G, K, V] =
       cmd.asInstanceOf[BaseRedis[F, K, V]].liftK[G]
   }
+
+  implicit class IMapKOps[F[_], K, V](val cmd: RedisCommands[F, K, V]) extends AnyVal {
+    def imapK[G[_]](fg: F ~> G, gf: G ~> F): RedisCommands[G, K, V] =
+      new BaseMapKRedisCommands[F, G, K, V](cmd, fg) with RedisCommands[G, K, V] {
+        override def transact[A](fs: TxStore[G, String, A] => List[G[Unit]]): G[Map[String, A]] =
+          fg(cmd.transact(store => fs(mapTxStore(store, fg)).map(gf.apply)))
+        override def transact_(fs: List[G[Unit]]): G[Unit] =
+          fg(cmd.transact_(fs.map(gf.apply)))
+        override def pipeline[A](fs: TxStore[G, String, A] => List[G[Unit]]): G[Map[String, A]] =
+          fg(cmd.pipeline(store => fs(mapTxStore(store, fg)).map(gf.apply)))
+        override def pipeline_(fs: List[G[Unit]]): G[Unit] =
+          fg(cmd.pipeline_(fs.map(gf.apply)))
+      }
+  }
+
+  implicit class InstrumentOps[F[_], K, V](val cmd: RedisCommands[F, K, V]) extends AnyVal {
+    def instrument(ff: F ~> F, includeTransactions: Boolean): RedisCommands[F, K, V] = {
+      if(includeTransactions)
```

Comment:
I'm not entirely sure whether both use cases actually exist or whether an opinionated version would be sufficient. Maybe someone has more insights here (I'm not using transactions/pipelines)?